src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1131,6 +1270,27 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
     }
     auto term = term_opt.value();
 
+    auto cfg = _metadata_cache.local().get_topic_cfg(model::tx_manager_nt);
+    if (!cfg) {
+        co_return init_tm_tx_reply{tx_errc::partition_not_exists};
+    }
+    if (stm->hosted_transactions_inited()) {
+        if (!stm->hosts(tx_id)) {
+            co_return tx_errc::not_coordinator;
+        }
+    } else {
+        auto hash_ranges_inited = co_await stm->try_init_hosted_transactions(
```

Comment:
it shouldn't use try_init_hosted_transactions, it's a dead lock (read & write locks, same fiber) just remove the else branch